dvf_csv/src/Plugin/Visualisation/Source/CsvFile.php

```diff
@@ -273,4 +273,24 @@ protected function getCacheKey() {
     return $plugin_id . ':' . $uri;
   }
 
+  /**
+   * Gets the cache expiry time from the DVF visualisation configuration.
+   *
+   * @return int
+   *   The time the cache object expires.
+   */
+  protected function getCacheExpiry() {
+    // Get the cache time set from visualisation.
```

Comment:
Considering these are the same could this not be moved to `Drupal\dvf\Plugin\Visualisation\Source\VisualisationSourceBase` as both this and JSON have the exact same?

Reply:
Thanks @rob3000 please see update.